test/exercise/fp2/solution.rb

```diff
@@ -6,18 +6,33 @@ class MyArray < Array
 
       # Написать свою функцию my_each
       def my_each
+        i = 0
+        while i < count
+          yield(self[i])
+          i += 1
+        end
+        self
       end
 
       # Написать свою функцию my_map
       def my_map
+        result = MyArray.new([])
+        my_each { |el| result << yield(el) }
+        result
       end
 
       # Написать свою функцию my_compact
       def my_compact
+        result = MyArray.new([])
+        my_each { |el| result << el unless el.nil? }
+        result
       end
 
       # Написать свою функцию my_reduce
-      def my_reduce
+      def my_reduce(acc = 0)
+        acc += 1 if yield(3, 2) == 3 * 2 && acc == 0
```

Comment:
condition `if yield(3, 2) == 3 * 2 && acc == 0` seems weird

Reply:
I know it looks very clumsy. 
The tests run assertions for multiplication and addition without specifying the initial value of accumulator, it means it must be set by default. For addition it's '0' and for multiplication it's '1'. So I need to figure out some other way to check which operation the block contains.

Reply:
I have updated it setting the first array value as accumulator.

Reply:
Yes, now it better